hnn_core/gui/gui.py

```diff
@@ -0,0 +1,1008 @@
+    viz_width = "1000px"
+    viz_height = "500px"
+    viz_window = Output(layout={
+        'height': viz_height,
+        'width': viz_width,
+    })
+
+    # header_button
+    header_button = HTML(value=f"""<div
+        style='background:{THEMECOLOR};text-align:center;color:white;'>
+        HUMAN NEOCORTICAL NEUROSOLVER</div>""")
+
+    # Simulation parameters
+    tstop = FloatText(value=170, description='tstop (ms):', disabled=False)
+    tstep = FloatText(value=0.025, description='tstep (ms):', disabled=False)
+    ntrials = IntText(value=1, description='Trials:', disabled=False)
+
+    # visualization layout
+    viz_layout_selection = Dropdown(
+        options=[('Horizontal', 'L-R'), ('Vertical', 'U-D')],
+        value='L-R', description='Layout:')
+    # initialize
+    initialize_viz_window(viz_window, variables, plot_outputs_list,
+                          plot_dropdowns_list, viz_width,
+                          viz_height, layout_option=viz_layout_selection.value,
+                          init=True)
+
+    # select backends
+    backend_selection = Dropdown(
+        options=[('Joblib', 'Joblib'), ('MPI', 'MPI')], value='Joblib',
+        description='Backend:')
+
+    mpi_cmd = Text(value='mpiexec', placeholder='Fill if applies',
+                   description='MPI cmd:', disabled=False)
+
+    joblib_cores = BoundedIntText(value=1, min=1,
+                                  max=multiprocessing.cpu_count(),
+                                  description='Cores:', disabled=False)
+
+    backend_config = Output()
+    handle_backend_change(backend_selection.value, backend_config, mpi_cmd,
+                          joblib_cores)
+
+    simulation_box = VBox([tstop, tstep, ntrials, backend_selection,
+                           backend_config])
+
+    # Sliders to change local-connectivity params
+    sliders = [
+        _get_sliders(params, [
+            'gbar_L2Pyr_L2Pyr_ampa', 'gbar_L2Pyr_L2Pyr_nmda',
+            'gbar_L2Basket_L2Pyr_gabaa', 'gbar_L2Basket_L2Pyr_gabab'
+        ]),
+        _get_sliders(params, [
+            'gbar_L2Pyr_L5Pyr', 'gbar_L2Basket_L5Pyr', 'gbar_L5Pyr_L5Pyr_ampa',
+            'gbar_L5Pyr_L5Pyr_nmda', 'gbar_L5Basket_L5Pyr_gabaa',
+            'gbar_L5Basket_L5Pyr_gabab'
+        ]),
+        _get_sliders(params,
+                     ['gbar_L2Pyr_L2Basket', 'gbar_L2Basket_L2Basket']),
+        _get_sliders(params, [
+            'gbar_L2Pyr_L5Basket', 'gbar_L5Pyr_L5Basket',
+            'gbar_L5Basket_L5Basket'
+        ])
+    ]
+
+    # accordians to group local-connectivity by cell type
+    boxes = [VBox(slider) for slider in sliders]
+    titles = [
+        'Layer 2/3 Pyramidal', 'Layer 5 Pyramidal', 'Layer 2 Basket',
+        'Layer 5 Basket'
+    ]
+    cell_connectivity = Accordion(children=boxes)
+    for idx, title in enumerate(titles):
+        cell_connectivity.set_title(idx, title)
+
+    # Dropdown for different drives
+    layout = Layout(width='200px', height='100px')
```

Comment:
can you give different names for the different widths and heights and move them in one place? E.g., `width_drive` etc. It will be easier to modify them in the future.

Reply:
We can do this later when refactoring the whole `gui.py`